pkg/network/usm/sharedlibraries/ebpf.go

```diff
@@ -96,39 +96,41 @@ func GetEBPFProgram(cfg *ddebpf.Config) *EbpfProgram {
 	progSingletonOnce.Do(func() {
 		progSingleton = &EbpfProgram{
 			cfg:     cfg,
-			libsets: make(map[Libset]*libsetData),
+			libsets: make(map[Libset]*libsetHandler),
+		}
+
+		// Initialize the libsets to avoid requiring a mutex on the map. Golang maps are thread safe
+		// for reads.
+		for libset := range LibsetToLibSuffixes {
+			progSingleton.libsets[libset] = &libsetHandler{
+				callbacks: make(map[*LibraryCallback]struct{}),
+			}
 		}
 	})
 	progSingleton.refcount.Inc()
 
 	return progSingleton
 }
 
-func (e *EbpfProgram) getLibsetData(libset Libset) *libsetData {
-	if data, ok := e.libsets[libset]; ok {
-		return data
-	}
-
-	data := &libsetData{
-		libset:    libset,
-		callbacks: make(map[*LibraryCallback]struct{}),
-		done:      make(chan struct{}),
-	}
-	e.libsets[libset] = data
-	return data
-}
-
+// enableLibset enables the eBPF program for the given libset. Assumes initMutex is locked
 func (e *EbpfProgram) enableLibset(libset Libset) {
-	// Force the program to be initialized if it's not already initialized
-	data := e.getLibsetData(libset)
-	data.enabled = true
+	e.libsets[libset].enabled = true
 }
 
+// isLibsetEnabled checks if the libset is enabled. Assumes initMutex is locked
 func (e *EbpfProgram) isLibsetEnabled(libset Libset) bool {
 	data, ok := e.libsets[libset]
 	return ok && data.enabled
 }
 
+// isLibsetEnabled checks if the libset has been requested to be enabled. Assumes initMutex is locked
+func (e *EbpfProgram) isLibsetRequested(libset Libset) bool {
+	data, ok := e.libsets[libset]
+	return ok && data.requested
+}
```

Comment:
didn't read all the change, and trying to review it as the average reader, why do we need to have "requested to be enabled" instead of taking the following assumption - If a libset exists in `libsets` it is requested?